community/kernel/src/main/java/org/neo4j/kernel/impl/index/labelscan/NativeLabelScanStore.java

```diff
@@ -321,9 +353,13 @@ private FileHandle storeFileHandle() throws IOException
         return single( pageCache.streamFilesRecursive( storeFile ).collect( Collectors.toList() ) );
     }
 
-    private void instantiateTree() throws IOException
+    /**
+     * @return true instantiated tree needs to be rebuilt
+     */
+    private boolean instantiateTree() throws IOException
     {
-        index = new GBPTree<>( pageCache, storeFile, new LabelScanLayout(), pageSize, treeMonitor(), NO_HEADER );
+        index = new GBPTree<>( pageCache, storeFile, new LabelScanLayout(), pageSize, treeMonitor(), getDirty );
+        return getDirty.isDirty();
```

Comment:
Does `getDirty` really need to be a field, of could you just create it here? I don't see you accessing it anywhere else.